### DISCUSSION TITLE
250 character Limit - How to get over it?

### DISCUSSION THREAD

**Discussion by andykaufseo** [coqui-ai/TTS#3548] [General Q&A]
for xtts v2, how do i change the 250 character limit? i know the api does this by itself, but i don't want to use the api as i need to pass some extra parameters. i can split the text into smaller chunks (sentences or paragraphs) but i need to generate 500 characters, how can i do that?

**Comment by barghavanii** [comment]
I guess if you meant the limitation of text prompt in hugging face demo which is (200 characters ) you can modify this changes by app.py --> line 200 you can make changes to 500 (if len(prompt) > 500:)

**Comment by TalonPower** [comment]
i tried making that change and it is still stopping at 200,  is it somewhere else in the code as well?

**Comment by barghavanii** [comment]
Hi ,
you can modify changes with tokenizer.py which you can find it in TTS/TTS/tts/layers/xtts/tokenizer.py
line 624 you can defined the limitation for text input .
Also in training.py that you pass the language and dataset and ... search for " max_text_length=200," and change it to your desirer but it is not recomended .
after all the above if your still get error of text length your dataset must be chunk of audio between 3-6 second and not more than 12 second if you intend to fine tune .

Best regards
Bahareh Arghavani Nobar
Graduate research assistant
SAIL LAB<https://sail-lab.org/>
University of New Haven
Email: ***@***.***
Tel: 860 944 5353
[cid:732fe70c-00c2-4538-91b8-ed2b7368b091]

​[https://ci3.googleusercontent.com/proxy/DkROAS12GWCwh0_8bxbCv6RlE4gMGP6HARzsNsA2KcSH0BycdFLGLeaOe4xIrHHuu7Ub2MNs7TSnzBaxYL-R32Ol5miKoswHnkzrCTseRr9h-S7aErKd5xwNqlVtl08jE15GNC3tTIgN=s0-d-e1-ft#http://orig13.deviantart.net/ab15/f/2017/047/9/f/signature_1_by_mehrjou-daz8smo.png]<http://thinkbeforeprinting.org/>

________________________________

**Comment by TalonPower** [comment]
ok so i forked the coqui tts then i changed the  tokenizer.py. 
https://github.com/TalonPower/TTS/blob/dev/TTS/tts/layers/xtts/tokenizer.py

 in the colab i changed the  requirements :https://github.com/TalonPower/xtts2-hf/blob/main/requirements.txt

to point at mine and still get same result.  the 200 limit.   also i made edits in my colab to change the look and feel and it still looks the same 

my app.py:
https://github.com/TalonPower/xtts2-hf/blob/main/app.py

**Comment by johndjchan** [comment]
anyone fixed the 250 cahracter limit issue pls
